core/capabilities/compute/compute.go

```diff
@@ -202,11 +247,27 @@ func (c *Compute) Close() error {
 	return nil
 }
 
-func NewAction(log logger.Logger, registry coretypes.CapabilitiesRegistry) *Compute {
+func NewAction(log logger.Logger, registry coretypes.CapabilitiesRegistry, handler *corecapabilities.OutgoingConnectorHandler) *Compute {
 	compute := &Compute{
-		log:      logger.Named(log, "CustomCompute"),
-		registry: registry,
-		modules:  newModuleCache(clockwork.NewRealClock(), 1*time.Minute, 10*time.Minute, 3),
+		log:                      logger.Named(log, "CustomCompute"),
+		registry:                 registry,
+		modules:                  newModuleCache(clockwork.NewRealClock(), 1*time.Minute, 10*time.Minute, 3),
+		outgoingConnectorHandler: handler,
 	}
 	return compute
 }
+
+func getMessageID(workflowID, workflowExecutionID string) (string, error) {
+	if err := validation.ValidateWorkflowOrExecutionID(workflowID); err != nil {
+		return "", fmt.Errorf("workflow ID %q is invalid: %w", workflowID, err)
+	}
+	if err := validation.ValidateWorkflowOrExecutionID(workflowExecutionID); err != nil {
+		return "", fmt.Errorf("workflow execution ID %q is invalid: %w", workflowExecutionID, err)
+	}
+	messageID := []string{
+		workflowID,
+		workflowExecutionID,
+		ghcapabilities.MethodComputeAction,
```

Comment:
I'd add a random string to this too (eg. uuid) so that a single execute call can make multiple requests without their responses overlapping